cmake/CMakeLists.txt

```diff
@@ -526,7 +526,21 @@ if(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
   find_package(Iconv REQUIRED)
   set(ICONV_LIB Iconv::Iconv)
 endif()
+
 find_package(Patch)
+if (WIN32 AND NOT Patch_FOUND)
+    # work around CI machines missing patch from the git install by falling back to the binary in this repo.
+    # replicate what happens in https://github.com/Kitware/CMake/blob/master/Modules/FindPatch.cmake but without
+    # the hardcoded suffixes in the path to the patch binary.
+    find_program(Patch_EXECUTABLE NAMES patch PATHS ${PROJECT_SOURCE_DIR}/external/git.Win32.2.41.03.patch)
```

Comment:
But, does this binary work fine on Windows ARM64 machines?

Reply:
in theory: https://learn.microsoft.com/en-us/windows/arm/overview#support-for-existing-windows-apps-on-arm. 

I'll test something on the QNN ARM64 CI to check

Reply:
Seems to work. https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1208745&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=cb33ee24-f492-54d2-727a-ae0a7bb84b2c is from running the patch.exe with `--version` and `--help` on the QNN ARM64 CI.

Reply:
Thank you!